modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1912,7 +1911,16 @@ void setViewportLength(double value) {
         this.absoluteOffset = getPosition() * (estimatedSize - viewportLength);
         recalculateEstimatedSize();
     }
-    double getViewportLength() {
+
+    /**
+     * Returns the length of the viewport portion of the {@code VirtualFlow} as computed during the layout pass.
+     * For a vertical flow, this corresponds to the height and for a horizontal flow to the width of the clip view,
+     * but it does not necessarily have to be the same value.
```

Comment:
@johanvos added a question in the CSR about this last  part:

> I understand and agree with the goal behind this.
> 
> I'm a bit confused though about the following: "...but it does not necessarily have to be the same value." -> can you elaborate a bit about this?

I share this concern. I think that removing that last clause and putting a period after "clip view" is probably the best.

Johan: what do you think?

Reply:
> @johanvos added a question in the CSR about this last part:
> 
> > I understand and agree with the goal behind this.
> > I'm a bit confused though about the following: "...but it does not necessarily have to be the same value." -> can you elaborate a bit about this?
> 
> I share this concern. I think that removing that last clause and putting a period after "clip view" is probably the best.
> 
> Johan: what do you think?

I agree that removing that clause is probably best to avoid confusion. Having open-ended suggestions in javadoc can lead to broad speculation, so I think it either should be explained (like you did in the CSR issue with the rubberband effect example) or removed.

Reply:
We should probably also avoid the word "corresponds".

````java
/**
 * Returns the length of the viewport portion of the {@code VirtualFlow} as computed during the layout pass.
 * For a vertical flow this is based on the height and for a horizontal flow on the width of the clip view.
 *
 * @return the viewport length in pixels
 * @since 23
 */
````
The text explains that depending on the orientation of the view height or width is used in the calculation and the word "based" makes it clear that the value can differ from the respective size of the view.

This version looks good to me. If it is fine for you too, I would check it in.

Reply:
I'm happy with this wording.